var/spack/repos/builtin/packages/py-kosh/package.py

```diff
@@ -28,3 +29,14 @@ class PyKosh(PythonPackage):
     depends_on("py-llnl-sina@1.11:", type=("build", "run"))
     depends_on("py-networkx", type=("build", "run"))
     depends_on("py-numpy", type=("build", "run"))
+
+    with when("@3:"):
+        depends_on("py-matplotlib")
+        depends_on("py-scipy")
+        depends_on("py-h5py@3:")
+        depends_on("py-scikit-learn@1.0.2:")
+        depends_on("py-pandas")
+        depends_on("py-hdbscan@0.8.29:")
+        depends_on("py-tqdm")
```

Comment:
These should all be `type=("build", "run")` if they are needed at run-time. Maybe also link if any of these are linked to. Also, if you sort these in the same order as setup.py, it becomes easier for me to review. Not required though.

Reply:
AFAIK none of them are linked in, the code that added these deps is all pure python.